colander/__init__.py

```diff
@@ -2120,6 +2120,8 @@ def clone(self):
         dictionaries are preserved."""
         children = [node.clone() for node in self.children]
         cloned = self.__class__(self.typ, *children)
+        cloned.children = []
+        _add_node_children(cloned, children)
```

Comment:
@disko that sounds like a bug in that test (relying on an internal detail / side effect?).  From the "external" perspective, the clone should have the desired list of child nodes either way.